packages/react-core/tsconfig.json

```diff
@@ -9,7 +9,7 @@
     "baseUrl": ".",
     "declaration": true,
     "esModuleInterop": true,
-    "jsx": "react-jsx",
+    "jsx": "react",
```

Comment:
Unfortunately we will have to switch back to old JSX transform because its ESM support only landed in 18 and the team did not backport it to older version <- https://github.com/facebook/react/pull/20304